tests/e2e.sh

```diff
@@ -70,16 +102,21 @@ set -e
 
 kubectl version
 
+# ensure that we build docker image in minikube
```

Comment:
Not sure whether I understand. In travis we don't have to switch docker to minikube's docker since it is shared anyway (it uses localkube) but in other setups (virtualbox for in my case, I need to build the docker image in minikube's context in order to run the ksm deployment.

Just to note,  you cannot use the command `minikube docker-env` command when running with `--vm-driver=none` as described [here](https://github.com/kubernetes/minikube/issues/2443).

As I see this, the solution I picked is the only one working :)

Reply:
I see. Thanks for the explanation. This is required when 
* no local docker daemon is already exists
* using minikube with none-none vm-drivers

Am i right?

Reply:
I am note sure whether I understand. There are 2 scenarios: a) run with --vm-driver=none and b) run with --vm-driver **!=none**.

a) In Travis, you run `minikube start --vm-driver=none` and It will setup the minikube locally without any virtualization. The kubernetes (minikube) and the system share the docker daemon. If you create a kubernetes Deployment with some docker image, it is tested, that the image is already on machine (unless `:latest` tag or `imagePullPolicy: Always` is used, but in this scenario it is not). Everything is easy and transparent.

b)
I locally don't want to run minikube on my host system (it is not recommended), so I run `minikube start --vm-driver=virtualbox` and it runs the kubernetes inside a virtualbox VM. But I run the kubernetes Deployment with the same image as before. Kubelet will not find the docker image in local docker storage (because it is a brand new virtualmachine), so docker/kubelet will try to pull it from docker registry. That is why I put the line with `minikube docker-env`)

```
$ minikube docker-env
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.100:2376"
export DOCKER_CERT_PATH="/home/tom/.minikube/certs"
export DOCKER_API_VERSION="1.23"
# Run this command to configure your shell:
# eval $(minikube docker-env)
```

If you run this, the docker client will use a remote docker server - the one running inside the VM where kubernetes is running. Therefore I run the command`eval` **before** building the image, I want the image to be present in docker storage **inside** the VM. After that the kubelet will use the docker image I just builded.

So we use the `minikube docker-env` only when `--vm-driver` is not `none`. So, you don't even have to have a docker daemon running, docker client binary is enough.

I hope I made it (my motivation & solution) more clear, if not, please ask again :)

Reply:
Excellent explanation.